src/main/java/com/bina/varsim/tools/evaluation/SAMcompare.java

```diff
@@ -358,5 +367,45 @@ public void setBedFilename(String bedFilename) {
         }
     }
 
+    /**
+     * check if a read is supposed to be aligned in the user-supplied BED region
+     * @param intersector BED-file object
+     * @param trueLoci collection of true alignment loci
+     * @param rec one alignment in SAM
+     * @return
+     */
+    private static boolean isContainedInBed(BedFile intersector, Collection<GenomeLocation> trueLoci, SAMRecord rec) {
+        if (intersector == null) {
+            return true;
+        }
+        boolean isContainedInBed = false;
+        for (GenomeLocation loc : trueLoci) {
+            if (intersector.contains(loc.chromosome, loc.location
+                    //here we should have used the true alignment length
+                    //however that is difficult to know unless we have a perfect BAM
+                    , loc.location + rec.getReadLength() - 1)) {
+                isContainedInBed = true;
+            }
+        }
+        return isContainedInBed;
+    }
 
+    /**
+     * calculate identity ratio of the read
+     * formula:
+     *
+     * sum of sequence matches (= or M in CIGAR) / total length of the read
+     *
+     * @param alignment
+     * @return
+     */
+    private static double getIdentity(SAMRecord alignment) {
+        int nMatches = 0;
+        for (CigarElement c : alignment.getCigar().getCigarElements()) {
+            if (ALL_POSSIBLE_MATCHES.contains(c.getOperator())) {
+                nMatches += c.getLength();
+            }
+        }
+        return (double) nMatches/alignment.getReadLength();
```

Comment:
This would penalize reads spanning indels/SVs unnecessarily even when they are aligned correctly with the right indels and softclips. For one, we should exclude softclipped bases in the denominator.